src/tcp/tcp.ts

```diff
@@ -295,6 +295,14 @@ module Tcp {
               if(this.state_ !== Connection.State.CLOSED) {
                 this.state_ = Connection.State.CONNECTED;
               }
+            })
+            .catch((err) => {
+              log.error('Failed to connect to %1, err=%2',
```

Comment:
So, this isn't quite the "log and re-throw" anti-pattern but generally I've found it better to delegate logging of catchable/recoverable errors like this to the caller. Let's remove the logging statement here -- presumably our callers are already listening for rejection.